### PR TITLE
[github] fix(ci): support for release branches and pre-release tags

### PR DESCRIPTION
**Pull request by diafour**
## Description

- Run build and test on push to release-* branches and tags with suffixes.
- No commenting for now, use curl to start e2e and deploy workflows for tags.

## Why do we need it, and what problem does it solve?

There is no way to release v1.29.0 without support for pre-release tags.

### CODE REVIEW

**Comment by diafour** [comment]
Currently tested in diafour/deckhouse

**Comment by diafour** [comment]
Ready for review.

**Review by nabokihms — COMMENTED** [review]
Please fill the changelog entry or, if you want to add nothing to the changelog, remove the template.

About the feature, everything looks acceptable to me. Let's wait for the @shvgn review before merging this.

**Review by shvgn — APPROVED** [review]
lgtm

File: .github/scripts/js/ci.js
Comment (diafour) [reply .github/scripts/js/ci.js:325]: Yeah, I'll change a comment.

Comment (diafour) [reply .github/scripts/js/ci.js:325]: I've moved a `replace` vs `replaceAll` explanation into README.md

**Review by nabokihms — APPROVED** [review]

**Pull request merged**
